docs/src/userguide/parallelism.rst

```diff
@@ -114,6 +114,13 @@ It currently supports OpenMP, but later on more backends might be supported.
         may give substantially different performance results, depending on the schedule, the load balance it provides,
         the scheduling overhead and the amount of false sharing (if any).
 
+    :param simd:
+        The ``simd`` argument indicates that the loop can be transformed into a SIMD loop (that is, multiple iterations
+        of the loop can be executed concurrently using SIMD instructions).
+        ``simd```is set to ``False`` by default.
+        Note that the use of ``simd`` typically requires ``@boundscheck(False)`` and ``@initializedcheck(False)``
```

Comment:
Why do these need to be disabled? (Could also be worth an explanation here.)

Reply:
Cython sometimes adds checks which require the GIL and Python objects. This simply doesn't work on a device.

I agree, this is not the best solution. Could the be related to the compilation context suggestion you made earlier?